pkg/standalone-utils/contracts/test/ProtocolFeeBurnerMock.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IProtocolFeeSweeper } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/IProtocolFeeSweeper.sol";
+import { IProtocolFeeBurner } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/IProtocolFeeBurner.sol";
+
+import { ERC20TestToken } from "@balancer-labs/v3-solidity-utils/contracts/test/ERC20TestToken.sol";
+
+contract ProtocolFeeBurnerMock is IProtocolFeeBurner {
+    using SafeERC20 for IERC20;
+
+    IProtocolFeeSweeper private _feeSweeper;
+
+    constructor(IProtocolFeeSweeper feeSweeper) {
+        _feeSweeper = feeSweeper;
+    }
+
+    /// @inheritdoc IProtocolFeeBurner
+    function burn(
+        address pool,
+        IERC20 feeToken,
+        uint256 feeTokenAmount,
+        IERC20 targetToken,
+        address recipient
+    ) external {
+        // Just emit the event, simulating the tokens being exchanged at 1-to-1.
+        feeToken.safeTransferFrom(address(_feeSweeper), address(this), feeTokenAmount);
```

Comment:
```suggestion
        feeToken.safeTransferFrom(msg.sender, address(this), feeTokenAmount);
```

If the burner is generically enough, msg.sender should be enough to get the fee sweeper's address. Then the burner wouldn't need to know about the _feeSweeper address.

Reply:
We could do this, but it means anyone could use these burners to do cow swap conversions from their own wallet. Is there a possible DoS attack from hundreds of accounts all calling these burner contracts?

Reply:
The way it's implemented, if the sweeper has tokens in the balance, has given the approval and the burn has failed for some reason, anyone could call burn and take the tokens to himself by using the recipient (since burn is async). Using msg.sender prevents that.

Reply:
Hanging approvals... have caused issues before. Thanks! Good catch.